discovery_child_development/notebooks/02_explore_concepts.py

```diff
@@ -11,164 +13,302 @@
 import altair as alt
 import numpy as np
 
+# project-specific imports
+from discovery_child_development.utils.cluster_analysis_utils import (
+    hdbscan_param_grid_search,
+    kmeans_param_grid_search,
+    highest_silhouette_model_params,
+)
+import discovery_child_development.utils.cluster_analysis_utils as cau
+
+# embeddings stuff
+import sentence_transformers
+from sentence_transformers import SentenceTransformer
+
+model = SentenceTransformer("all-MiniLM-L6-v2")
+
+# %%
+# Set variables
+
+SEED = 42
+
 load_dotenv()
 S3_BUCKET = os.environ["S3_BUCKET"]
 s3_client = boto3.client("s3")
 
-CONCEPTS = "C109260823|C2993937534|C2777082460|C2911196330|C2993037610|C2779415726|C2781192327|C15471489|C178229462"
-
-data_path = f"inputs/data/openAlex/concepts/concepts_metadata_{CONCEPTS}_year-2023.csv"
+CONCEPT_IDS = [
+    "C109260823",  # child development
+    "C2993937534",  # childhood development
+    "C2777082460",  # early childhood
+    "C2911196330",  # child rearing
+    "C2993037610",  # child care
+    "C2779415726",  # child protection
+    "C2781192327",  # child behavior checklist
+    "C15471489",  # child psychotherapy
+    "C178229462",  # early childhood education
+    # "C138496976",  # developmental psychology (level 1).
+]
+
+CONCEPTS = "|".join(CONCEPT_IDS)
+
+YEARS = [2019, 2020, 2021, 2022, 2023]
+years_list = [str(x) for x in YEARS]
+
+# %%
+# Load concepts metadata
+data_path = f"data/openAlex/concepts/concepts_metadata_{CONCEPTS}_year-{'-'.join(years_list)}.csv"
 response = s3_client.get_object(Bucket=S3_BUCKET, Key=data_path)
 csv_data = response["Body"].read()
 concepts_df = pd.read_csv(BytesIO(csv_data), index_col=0)
 concepts_df.head()
```

Comment:
Probably better to use nesta_ds_utils, or factor out into your own module for loading csv files from s3?

Reply:
Yes, good point